diff --git a/web/modules/contrib/language_access/src/EventSubscriber/LanguageAccessSubscriber.php b/web/modules/contrib/language_access/src/EventSubscriber/LanguageAccessSubscriber.php
index 3b7ed39..2672497 100644
--- a/web/modules/contrib/language_access/src/EventSubscriber/LanguageAccessSubscriber.php
+++ b/web/modules/contrib/language_access/src/EventSubscriber/LanguageAccessSubscriber.php
@@ -72,23 +72,26 @@ class LanguageAccessSubscriber implements EventSubscriberInterface {
     // Allow user path.
     if (strpos($requestUrl, '/user/') === FALSE) {
       // Check access to language.
+      $route_match = \Drupal::routeMatch();
+      $current_route = $route_match->getRouteName();
       if (!$this->currentUser->hasPermission('access language ' . $language->getId())) {
-        $default_language = $this->languageManager->getDefaultLanguage();
-        // We still want to allow access to default language.
-        if ($language->getId() !== $default_language->getId()) {
-          // Do not execute on drush.
-          if (PHP_SAPI != 'cli') {
-            // Display the default access denied page.
-            if ($event->getRequestType() === HttpKernelInterface::MASTER_REQUEST) {
-              throw new AccessDeniedHttpException();
-            }
+        var_dump($current_route);
+        $patterns = [
+          '/^entity\.([a-za-zA-Z0-9_\-.]+)$/',
+          '/^webcomposer_config([a-zA-Z0-9_\-\.]+)$/',
+          '/^system.admin_config$/'
+        ];
+
+        foreach ($patterns as $pattern) {
+          if (preg_match($pattern, $current_route)) {
+            throw new AccessDeniedHttpException();
           }
         }
-        else {
-          $route_match = \Drupal::routeMatch();
-          $current_route = $route_match->getRouteName();
-          $pattern = '/^entity\.([a-za-zA-Z0-9_\-]+)\.edit_form$/';
-          if (preg_match($pattern, $current_route, $matches)) {
+      } else {
+        // fix for menu entities
+        if ($current_route === 'entity.menu.edit_form') {
+          $langcode = $route_match->getParameter('menu')->get('langcode');
+          if (!$this->currentUser->hasPermission('access language ' . $langcode)) {
             throw new AccessDeniedHttpException();
           }
         }
